### ISSUE TITLE
Trusted peer responded with different node id

### ISSUE THREAD

**Issue by dorin100** [input-output-hk/jormungandr#965]
**Describe the bug**
Seeing logs saying that the trusted peers are responding with different node ids (looking at the node id values, they seem the same).

**Mandatory Information**
- using release files for windows - `v0.6.1`

**To Reproduce**
1. Start the node with the provided trusted peers 

**Expected behavior**

**Additional context**
- node log:
```
Oct 16 06:38:11.694 INFO peer responded with different node id: 8091571e2ae4ba01672692460785315ca2f16dc3288b324a3931f7712a45e309, node_id: 8091571e2ae4ba01672692460785315ca2f16dc3288b324a3931f7712a45e309, peer_addr: 54.153.19.202:3000, task: network
```

**Comment by bobdobs** [comment]
Someone on Telegram suggested that this happens when the peer has a floating IP config

**Comment by mzabaluev** [comment]
There's been a logging problem: the condition is real, but one of the IDs was wrongly logged in both places. Fixed in #981.

**Comment by mzabaluev** [comment]
This happens when a peer changes its ID, but is contacted by its old ID on the same IP address based on outdated gossip or trusted peer configuration. We are debating if this should terminate the connection; I think it will need to be if/when gossip and subscription IDs are authenticated and connection integrity is protected with TLS.
